src/app/api/workflow-states/workflowStates.service.ts

```diff
@@ -11,14 +11,33 @@ class WorkflowStatesService extends BaseService {
     const policyGate = new PoliciesService(this.user)
     policyGate.authorize(UserAction.Read, Resource.Tasks)
 
-    const filters = { where: { workspaceId: this.user.workspaceId } }
-    let workflowStates = await this.db.workflowState.findMany(filters)
+    let workflowStates: WorkflowStateResponse[] = await this.db.$queryRaw`
+    SELECT * FROM "WorkflowStates"
+    WHERE "workspaceId" = ${this.user.workspaceId}
+    ORDER BY 
+      CASE 
+        WHEN "key" = 'todo' THEN 1
+        WHEN "key" = 'inProgress' THEN 2
+        WHEN "key" = 'completed' THEN 3
+        ELSE 4
+      END;
+  `
 
     if (!workflowStates.length) {
       await this.setDefaultWorkflowStates()
       // Query workflowStates again because createMany only returns count and not an array of rows
       // also autogenerated ID field needs to be queried again anyway
-      workflowStates = await this.db.workflowState.findMany(filters)
+      workflowStates = await this.db.$queryRaw`
+      SELECT * FROM "WorkflowStates"
+      WHERE "workspaceId" = ${this.user.workspaceId}
+      ORDER BY 
+        CASE 
+          WHEN "key" = 'todo' THEN 1
+          WHEN "key" = 'inProgress' THEN 2
+          WHEN "key" = 'completed' THEN 3
+          ELSE 4
+        END;
+    `
```

Comment:
Same thing about workflow state type. Also please make this query reusable since this is used in more than one place.